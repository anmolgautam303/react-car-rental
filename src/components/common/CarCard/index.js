import React from 'react';
import moment from 'moment';

export const CarCard = props => {
  let days = 1;
  if (props.rentedDates.to !== undefined) {
    days = moment(props.rentedDates.to).diff(moment(props.rentedDates.from), 'days') + 1
  }

  return (
    <div className="col-lg-4 col-md-6 col-sm-12">
      <div className="balance-box bg-light">
        <header className="car-header">
          <h6 className="category">{props.item.type}</h6>

          <div className="row">
            <div className="col-sm">
              <h6>{props.item.name}</h6>
            </div>
            <div className="col-sm">
              <h6 className="price">{(props.item.price * days).toFixed(2)}â‚¬</h6>
            </div>
          </div>
        </header>

        <figure className="car-container image-placeholder">
          <img
            className="responsive-image car-image" 
            src={require(`../../../assets/img/cars/${props.item.image}`)}
            alt="car"
          />
        </figure>

        {props.children}
      </div>
    </div>

  );
};