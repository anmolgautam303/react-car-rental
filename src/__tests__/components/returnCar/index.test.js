import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

import { CarCard } from '../../../components/common/CarCard';
import ReturnCard from '../../../components/returnCar';

import CarImage from "../../../assets/img/cars/bmw-7-series.jpg"

configure({ adapter: new Adapter() })

describe('Car Card', () => {
  let props = { 
    rentedDates: {
      "from": "Wed May 29 2019 12:00:00 GMT+0300",
      "to": "Thu May 30 2019 12:00:00 GMT+0300"
    },
    "rentedCard": {
      "id": 1,
      "name": "BMW 7 Series",
      "type": "LARGE",
      "price": 618.77,
      "seats": "5",
      "transmission": "Automatic",
      "engine": "Diesel",
      "image": "bmw-7-series.jpg"
    }
  };
  let returnCard = shallow(<ReturnCard {...props} />);

  it('renders properly', () => {
    expect(returnCard).toMatchSnapshot();
  });

  it('displays the correct car type', () => {
    expect(returnCard.find('h3').text()).toEqual('Return car');
  });

  it('displays the correct price', () => {
    expect(returnCard.find('h5').text()).toEqual('Return date: 30-05-2019');
  });
});