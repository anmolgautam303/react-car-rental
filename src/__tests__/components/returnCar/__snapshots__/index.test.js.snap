// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car Card renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReturnCar
    rentedCard={
      Object {
        "engine": "Diesel",
        "id": 1,
        "image": "bmw-7-series.jpg",
        "name": "BMW 7 Series",
        "price": 618.77,
        "seats": "5",
        "transmission": "Automatic",
        "type": "LARGE",
      }
    }
    rentedDates={
      Object {
        "from": "Wed May 29 2019 12:00:00 GMT+0300",
        "to": "Thu May 30 2019 12:00:00 GMT+0300",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <h3>
          Return car
        </h3>
        <div
          className="row return-card-center"
        >
          <CarCard
            item={
              Object {
                "engine": "Diesel",
                "id": 1,
                "image": "bmw-7-series.jpg",
                "name": "BMW 7 Series",
                "price": 618.77,
                "seats": "5",
                "transmission": "Automatic",
                "type": "LARGE",
              }
            }
            rentedDates={
              Object {
                "from": "Wed May 29 2019 12:00:00 GMT+0300",
                "to": "Thu May 30 2019 12:00:00 GMT+0300",
              }
            }
          >
            <div
              className="button-container"
            >
              <h5
                className="return-date"
              >
                Return date: 
                30-05-2019
              </h5>
            </div>
          </CarCard>
        </div>
      </div>,
      "className": "return-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Return car
          </h3>,
          <div
            className="row return-card-center"
          >
            <CarCard
              item={
                Object {
                  "engine": "Diesel",
                  "id": 1,
                  "image": "bmw-7-series.jpg",
                  "name": "BMW 7 Series",
                  "price": 618.77,
                  "seats": "5",
                  "transmission": "Automatic",
                  "type": "LARGE",
                }
              }
              rentedDates={
                Object {
                  "from": "Wed May 29 2019 12:00:00 GMT+0300",
                  "to": "Thu May 30 2019 12:00:00 GMT+0300",
                }
              }
            >
              <div
                className="button-container"
              >
                <h5
                  className="return-date"
                >
                  Return date: 
                  30-05-2019
                </h5>
              </div>
            </CarCard>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Return car",
          },
          "ref": null,
          "rendered": "Return car",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CarCard
              item={
                Object {
                  "engine": "Diesel",
                  "id": 1,
                  "image": "bmw-7-series.jpg",
                  "name": "BMW 7 Series",
                  "price": 618.77,
                  "seats": "5",
                  "transmission": "Automatic",
                  "type": "LARGE",
                }
              }
              rentedDates={
                Object {
                  "from": "Wed May 29 2019 12:00:00 GMT+0300",
                  "to": "Thu May 30 2019 12:00:00 GMT+0300",
                }
              }
            >
              <div
                className="button-container"
              >
                <h5
                  className="return-date"
                >
                  Return date: 
                  30-05-2019
                </h5>
              </div>
            </CarCard>,
            "className": "row return-card-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <div
                className="button-container"
              >
                <h5
                  className="return-date"
                >
                  Return date: 
                  30-05-2019
                </h5>
              </div>,
              "item": Object {
                "engine": "Diesel",
                "id": 1,
                "image": "bmw-7-series.jpg",
                "name": "BMW 7 Series",
                "price": 618.77,
                "seats": "5",
                "transmission": "Automatic",
                "type": "LARGE",
              },
              "rentedDates": Object {
                "from": "Wed May 29 2019 12:00:00 GMT+0300",
                "to": "Thu May 30 2019 12:00:00 GMT+0300",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h5
                  className="return-date"
                >
                  Return date: 
                  30-05-2019
                </h5>,
                "className": "button-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Return date: ",
                    "30-05-2019",
                  ],
                  "className": "return-date",
                },
                "ref": null,
                "rendered": Array [
                  "Return date: ",
                  "30-05-2019",
                ],
                "type": "h5",
              },
              "type": "div",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <h3>
            Return car
          </h3>
          <div
            className="row return-card-center"
          >
            <CarCard
              item={
                Object {
                  "engine": "Diesel",
                  "id": 1,
                  "image": "bmw-7-series.jpg",
                  "name": "BMW 7 Series",
                  "price": 618.77,
                  "seats": "5",
                  "transmission": "Automatic",
                  "type": "LARGE",
                }
              }
              rentedDates={
                Object {
                  "from": "Wed May 29 2019 12:00:00 GMT+0300",
                  "to": "Thu May 30 2019 12:00:00 GMT+0300",
                }
              }
            >
              <div
                className="button-container"
              >
                <h5
                  className="return-date"
                >
                  Return date: 
                  30-05-2019
                </h5>
              </div>
            </CarCard>
          </div>
        </div>,
        "className": "return-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Return car
            </h3>,
            <div
              className="row return-card-center"
            >
              <CarCard
                item={
                  Object {
                    "engine": "Diesel",
                    "id": 1,
                    "image": "bmw-7-series.jpg",
                    "name": "BMW 7 Series",
                    "price": 618.77,
                    "seats": "5",
                    "transmission": "Automatic",
                    "type": "LARGE",
                  }
                }
                rentedDates={
                  Object {
                    "from": "Wed May 29 2019 12:00:00 GMT+0300",
                    "to": "Thu May 30 2019 12:00:00 GMT+0300",
                  }
                }
              >
                <div
                  className="button-container"
                >
                  <h5
                    className="return-date"
                  >
                    Return date: 
                    30-05-2019
                  </h5>
                </div>
              </CarCard>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Return car",
            },
            "ref": null,
            "rendered": "Return car",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <CarCard
                item={
                  Object {
                    "engine": "Diesel",
                    "id": 1,
                    "image": "bmw-7-series.jpg",
                    "name": "BMW 7 Series",
                    "price": 618.77,
                    "seats": "5",
                    "transmission": "Automatic",
                    "type": "LARGE",
                  }
                }
                rentedDates={
                  Object {
                    "from": "Wed May 29 2019 12:00:00 GMT+0300",
                    "to": "Thu May 30 2019 12:00:00 GMT+0300",
                  }
                }
              >
                <div
                  className="button-container"
                >
                  <h5
                    className="return-date"
                  >
                    Return date: 
                    30-05-2019
                  </h5>
                </div>
              </CarCard>,
              "className": "row return-card-center",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <div
                  className="button-container"
                >
                  <h5
                    className="return-date"
                  >
                    Return date: 
                    30-05-2019
                  </h5>
                </div>,
                "item": Object {
                  "engine": "Diesel",
                  "id": 1,
                  "image": "bmw-7-series.jpg",
                  "name": "BMW 7 Series",
                  "price": 618.77,
                  "seats": "5",
                  "transmission": "Automatic",
                  "type": "LARGE",
                },
                "rentedDates": Object {
                  "from": "Wed May 29 2019 12:00:00 GMT+0300",
                  "to": "Thu May 30 2019 12:00:00 GMT+0300",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h5
                    className="return-date"
                  >
                    Return date: 
                    30-05-2019
                  </h5>,
                  "className": "button-container",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Return date: ",
                      "30-05-2019",
                    ],
                    "className": "return-date",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Return date: ",
                    "30-05-2019",
                  ],
                  "type": "h5",
                },
                "type": "div",
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
